strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetArtifactGenerator.java

```diff
@@ -24,88 +29,123 @@
-            "<Relationship Type=\"http://schemas.microsoft.com/packaging/2010/07/manifest\" Target=\"/%s.nuspec\" Id=\"Rc20b205c579d4f85\" />\r\n  " +
-            "<Relationship Type=\"http://schemas.openxmlformats.org/package/2006/relationships/metadata/core-properties\" Target=\"/package/services/metadata/core-properties/metadata.psmdcp\" Id=\"R23f62\r\ne2778b3442e\" />\r\n"  +
-            "</Relationships>";
-    
-    private String basedir;
-
-    public NugetPackageGenerator()
+                                               "<Relationships xmlns=\"http://schemas.openxmlformats.org/package/2006/relationships\">\r\n  " +
+                                               "<Relationship Type=\"http://schemas.microsoft.com/packaging/2010/07/manifest\" Target=\"/%s.nuspec\" Id=\"Rc20b205c579d4f85\" />\r\n  " +
+                                               "<Relationship Type=\"http://schemas.openxmlformats.org/package/2006/relationships/metadata/core-properties\" Target=\"/package/services/metadata/core-properties/metadata.psmdcp\" Id=\"R23f62\r\ne2778b3442e\" />\r\n" +
+                                               "</Relationships>";
+
+    private Path basedir;
+
+    public NugetArtifactGenerator()
+    {
+    }
+
+    public NugetArtifactGenerator(String basedir)
     {
+        this.basedir = Paths.get(basedir);
     }
 
-    public NugetPackageGenerator(String basedir)
+    public NugetArtifactGenerator(Path basedir)
     {
         this.basedir = basedir;
     }
 
-    public NugetPackageGenerator(File basedir)
+    @Override
+    public Path generateArtifact(String id,
+                                 String version,
+                                 int size)
+            throws IOException
     {
-        this.basedir = basedir.getAbsolutePath();
+        // Use of size is not implemented
+        return generateArtifact(id, version);
     }
 
-    public void generateNugetPackage(String id,
-                                     String versionString,
-                                     String... dependencyList)
-            throws NugetFormatException, JAXBException, IOException, NoSuchAlgorithmException
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
+            throws IOException
     {
-        File file = new File(getBasedir(), String.format("%s/%s/%s.%s.nupkg", id, versionString, id, versionString));
+        // Use of size is not implemented
+        NugetArtifactCoordinates nac = NugetArtifactCoordinates.parse(uri.toString());
 
-        //noinspection ResultOfMethodCallIgnored
-        file.getParentFile().mkdirs();
+        return generateArtifact(nac.getId(), nac.getVersion());
+    }
 
-        SemanticVersion version = SemanticVersion.parse(versionString);
-        logger.debug("Version of the nupkg package: ", version.toString());
-        
-        generate(file, id, version, dependencyList);
+    private Path generateArtifact(String id,
```

Comment:
you can make it public for not to use `generateArtifact(packageId, packageVersion, 0)` method calls below